src/config/machines/example-grant-machine.js

```diff
@@ -0,0 +1,108 @@
+import { assign, createMachine, interpret } from "xstate";
+
+// Define the implementations for your actions
+const actionImplementations = {
+  trackPageCompletion: assign({
+    /**
+     * Updates the completed pages
+     * @param {object} context machine context
+     * @param {object} event triggered event
+     * @returns {string} event's question
+     */
+    completedPageIds: (context, event) => {
+      return [...context.completedPageIds, context.currentPageId];
+    },
+  }),
+  updateCurrentPageId: assign({
+    /**
+     * Updates the current page
+     * @param {object} context machine context
+     * @param {object} event triggered event
+     * @returns {string} event's question
+     */
+    currentPageId: (context, event) => {
+      return event.nextPageId;
+    },
+  }),
+};
+
+const exampleGrantMachine = createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5RgB4EMC2AHANmA4gE5oB2ALgLJoDGAFgJYlgB0sZahZAxAHICiADQAqAbQAMAXUSgsAe1j0y9WSWkgUiAIwBmAKzMAbJoCcAdgAcB0wd0AWIwdsAaEAE8t2sc1vaATNs1bWzFNczFfCwBfSJdUTFwCYnIqOkYWADN6QjYAWgBHAFc4JRUuACEAQQBhAGlxKSQQOQUS1UaNBH8DZl9jX10Q7SNbHV8XdwRzTWZQ0Ls562MxXWjY9Gw8IlJKGgYmZkzssnyitmUSXkFRSTVmxXO1DttjcxnjXVNfTVMgvXNjbTjDy+ZgA-yBYKhcLmVbgdYJLbJXZpVhgagqCAnYrncrVOo3Rp3VqPRC6F7MSzmezmPwRMQ+IGTbSGKZ9Wx2UxiPS6AywuIbRLbFJ7FiwNEYrFnUr8YT1W7ye4qEmTMLMMTmMKaQLGAxU3xjNxaCLMPSadWjek-XTaaIxEAkWQQOBqfkIpI7VJMeUtB7tRA5AyMgN8+Gbd3ClFsDhkb2KtqgJ4GiaaD4Ul5U61iPrmTkwu2usNC5H7Q65QrYpWEhXEv0IbTaYzMXS6cw88y9YxmRyMzQGLwNjXszzZ0yaEPxQtIz2i8UkTHlqXxprV30JxCckE6dlciIa0xDHsGEH-QeZkdj-OhwVTkUHRhoHCxmtrhBmqlq3xWAFcnQfQ-TGws07IITH6FZbSAA */
```

Comment:
Would be a good to load these from a file per grant eventually....

Reply:
Definitely in a future story in iritation 2 most likely together with machine per user session config.